pkg/spoke/addon/configuration.go

```diff
@@ -34,6 +34,11 @@ type registrationConfig struct {
 	addOnAgentRunningOutsideManagedCluster bool
 }
 
+type addonInstallOption struct {
+	InstallationNamespace             string
+	AgentRunningOutsideManagedCluster bool
```

Comment:
should these have json marker? also in registrationConfig, we should put it in the separate struct so we do not duplicate it, right?

Reply:
Do you mean we put addonInstallOption inline in registrationConfig, right？